### PR TITLE
Add GKE config-connector example

### LINKED ISSUE #125: Improve GKE example/test coverage
We should significantly improve our GKE test coverage to cover the basic GKE use cases. In addition, some coverage of more advanced integrations should be considered, e.g.
1. Using workload identities: https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity
2. Service account/IAM integration: https://cloud.google.com/kubernetes-engine/docs/how-to/iam
3. Advanced use cases like shared VPCs: https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-shared-vpc
4. Model some best practices such as https://cloud.google.com/kubernetes-engine/docs/best-practices/enterprise-multitenancy

### PR DESCRIPTION
**Pull request by lblackstone**
Based on https://cloud.google.com/config-connector/docs/how-to/install-upgrade-uninstall

~I haven't yet enabled this in CI until #175 is done to avoid deleting IAM bindings.~

This example currently requires manual steps, so I'm not going to add it to the CI tests.

Fix #125

### CODE REVIEW

File: examples/gke-cloud-connector-ts/index.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        token-key: '{.credential.access_token}'
+      name: gcp
+`;
+});
+
+// Create a Provider to use the new kubeconfig.
+const k8sProvider = new k8s.Provider("k8s", {kubeconfig});
+
+// Create a random suffix for the ServiceAccount.
+const saSuffix = new random.RandomPet("saSuffix", {length: 1}).id;
```
Comment (viveklak) [inline examples/gke-cloud-connector-ts/index.ts:82]: Did you intend for this to be a RandomString instead?

**Review by viveklak — APPROVED** [review]

**Review by mikhailshilkov — APPROVED** [review]
LGTM. Since we aren't using this in our CI tests, should we move it to the examples repo instead?

File: examples/gke-cloud-connector-ts/index.ts (hunk 1 above)
Comment (lblackstone) [reply examples/gke-cloud-connector-ts/index.ts:82]: No, I used pet because it was a little easier to find in the console when I was poking around.

**Comment by lblackstone** [comment]
> LGTM. Since we aren't using this in our CI tests, should we move it to the examples repo instead?

I think that makes sense at some point, but considering that it still has a couple usability TODOs, maybe we should leave it here for now? In other words, it's _possible_ to accomplish this today, but it's still a bit rough.

**Comment by mikhailshilkov** [comment]
> I think that makes sense at some point, but considering that it still has a couple usability TODOs, maybe we should leave it here for now?

My thinking is that we choose the place based on the value of an example:

1. If it brings us confidence in quality, it should be a CI test.
2. If it shows users how to achieve something, it should be in examples.
3. 
4. Is there a third option? Do you want to file an issue that we need to resolve to get to (1) and/or (2)?

**Comment by viveklak** [comment]
> > I think that makes sense at some point, but considering that it still has a couple usability TODOs, maybe we should leave it here for now?
> 
> My thinking is that we choose the place based on the value of an example:
> 
> 1. If it brings us confidence in quality, it should be a CI test.
> 2. If it shows users how to achieve something, it should be in examples.
> 
> Is there a third option? Do you want to file an issue that we need to resolve to get to (1) and/or (2)?

I agree with this thinking and I think it sits within (2) so moving to examples seems reasonable. I think we have issues open for all the major rough edges so it seems reasonable to me.

**Comment by lblackstone** [comment]
Sounds good. I'll move to examples.

**Comment by lblackstone** [comment]
Closing in favor of https://github.com/pulumi/examples/pull/1064

**Pull request closed without merging**